orderbook/src/database/trades.rs

```diff
@@ -27,8 +27,15 @@ impl Database {
                 t.sell_amount - t.fee_amount as sell_amount_before_fees,\
                 o.owner, \
                 o.buy_token, \
-                o.sell_token \
+                o.sell_token, \
+                s.tx_hash \
             FROM trades t \
+            LEFT OUTER JOIN settlements s \
+            ON t.block_number = s.block_number
+            AND t.log_index < (
+                SELECT MIN(log_index) FROM settlements s2
+                WHERE t.block_number = s2.block_number
+            )
```

Comment:
This turned out more complicated than I thought:

```suggestion
            LEFT OUTER JOIN LATERAL \
                ( \
                    SELECT tx_hash \
                    FROM settlements s \
                    WHERE \
                        s.block_number = t.block_number AND \
                        s.log_index > t.log_index \
                    ORDER BY s.log_index ASC \
                    LIMIT 1 \
                ) settlement \
                ON true \
```

see https://www.postgresql.org/docs/13/queries-table-expressions.html ctr+f for `lateral`.